### PR TITLE
[next] changed condition for when to set rscDidPostponeHeader and added tests

### PR DESCRIPTION
**Pull request by wyattjoh**
This adds some tests to the PPR implementation for Next.js. This also fixes a bug where the static pages were incorrectly generating a header that falsly indicated that it postponed.

### CODE REVIEW

**Comment by wyattjoh** [comment]
Current dependencies on/for this PR:
* main
  * **PR #10808** <a href="https://app.graphite.dev/github/pr/vercel/vercel/10808" target="_blank">Graphite</a>  👈

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/vercel/vercel/10808?utm_source=stack-comment).

File: packages/next/src/utils.ts [hunk 1]
```diff
@@ -2429,7 +2429,8 @@ export const onPrerenderRoute =
                 initialHeaders: {
                   'content-type': rscContentTypeHeader,
                   vary: rscVaryHeader,
-                  ...(experimentalPPR && rscDidPostponeHeader
+                  // If it contains a pre-render, then it was postponed.
+                  ...(prerender && rscDidPostponeHeader
```
Comment (wyattjoh) [inline packages/next/src/utils.ts:2433]: This ensures that the prerender is only generated with the "didPostponeHeader" being set if indeed that specific prerender did actually postpone. This is indicated by the truthyness of the `prerender` variable here.

File: test/lib/deployment/test-deployment.js [hunk 2]
```diff
@@ -283,7 +283,9 @@ async function runProbe(probe, deploymentId, deploymentUrl, ctx) {
       }
     });
     hadTest = true;
-  } else if (probe.notResponseHeaders) {
+  }
+
+  if (probe.notResponseHeaders) {
```
Comment (wyattjoh) [inline test/lib/deployment/test-deployment.js:288]: Unrelated but useful, this just ensures that if the probe specified both `responseHeaders` AND `notResponseHeaders` it checks both of them.

File: packages/next/test/fixtures/00-app-dir-ppr-full/index.test.js [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+  { pathname: '/on-demand/a', dynamic: true },
+  { pathname: '/on-demand/b', dynamic: true },
+  { pathname: '/on-demand/c', dynamic: true },
+  { pathname: '/static', dynamic: false },
+  { pathname: '/no-suspense', dynamic: true },
+  { pathname: '/no-suspense/nested/a', dynamic: true },
+  { pathname: '/no-suspense/nested/b', dynamic: true },
+  { pathname: '/no-suspense/nested/c', dynamic: true },
+  // TODO: uncomment when we've fixed the 404 case for force-dynamic pages
+  // { pathname: '/dynamic/force-dynamic', dynamic: 'force-dynamic' },
```
Comment (wyattjoh) [inline packages/next/test/fixtures/00-app-dir-ppr-full/index.test.js:20]: This will be addressed in a follow up PR as it requires changes to Next.js.

**Review by ztanner — APPROVED** [review]

File: packages/next/test/fixtures/00-app-dir-ppr-full/next.config.js [hunk 4]
```diff
@@ -0,0 +1,14 @@
+/** @type {import('next').NextConfig} */
+const nextConfig = {
+  experimental: {
+    ppr: true,
+  },
+  eslint: {
+    // Warning: This allows production builds to successfully complete even if
+    // your project has ESLint errors.
+    ignoreDuringBuilds: true,
+  },
```
Comment (Ethan-Arrowood) [inline packages/next/test/fixtures/00-app-dir-ppr-full/next.config.js:10]: Can we remove eslint from the fixture? or is it baked into Next.js?

**Review by Ethan-Arrowood — APPROVED** [review]
tests look good to me. Would appreciate some review from Next.js team too.

File: packages/next/test/fixtures/00-app-dir-ppr-full/next.config.js (hunk 4 above)
Comment (wyattjoh) [reply packages/next/test/fixtures/00-app-dir-ppr-full/next.config.js:10]: It's baked in to Next.js's build process.

**Review by ijjk — APPROVED** [review]

**Pull request merged**